aws/src/main/java/org/apache/iceberg/aws/s3/S3FileIO.java

```diff
@@ -116,16 +118,19 @@ public void initialize(Map<String, String> properties) {
       this.s3 = AwsClientFactories.from(properties)::s3;
     }
 
-    // Report Hadoop metrics if Hadoop is available
-    try {
-      DynConstructors.Ctor<MetricsContext> ctor =
-          DynConstructors.builder(MetricsContext.class).hiddenImpl(DEFAULT_METRICS_IMPL, String.class).buildChecked();
-      this.metrics = ctor.newInstance("s3");
+    String metricsImpl = properties.getOrDefault(
+            CatalogProperties.IO_METRICS_IMPL, CatalogProperties.IO_METRICS_IMPL_DEFAULT);
 
-      metrics.initialize(properties);
-    } catch (NoSuchMethodException | ClassCastException e) {
-      LOG.warn("Unable to load metrics class: '{}', falling back to null metrics", DEFAULT_METRICS_IMPL, e);
+    Map<String, String> metricContextProperties = properties;
+    if (!properties.containsKey(CatalogProperties.IO_METRICS_SCHEME)) {
+      metricContextProperties = ImmutableMap.<String, String>builder()
+              .putAll(properties)
+              .put(CatalogProperties.IO_METRICS_SCHEME, IO_METRICS_SCHEME_S3)
+              .build();
     }
+
+    this.metrics = CatalogUtil.loadFileIOMetricsContext(
```

Comment:
this overrides the default `MetricsContext.nullMetrics()` that gets set in the variable initialization win will result in NPE (see other note in catalog util)